mpl_data_cast/gui/main.py

```diff
@@ -74,8 +74,9 @@ def __init__(self, *args, **kwargs):
         recipes = mpldc_recipe.get_available_recipe_names()
         for rr in recipes:
             self.comboBox_recipe.addItem(rr, rr)
-        # Set default recipe to "CatchAll"
-        default = recipes.index("CatchAll")
+        # Set recipe according to preferences
+        default = recipes.index(self.settings.value("main/recipe",
```

Comment:
When the string in the recipes is not in recipes, you would get a KeyError here. Please change to sth like:
```python
recipe_settings = self.settings.value...
if recipe_settings in recipes:
    default = ..
else:
    default = recipes.index("CatchAll")
```

Reply:
Makes sense if the available recipes change e.g. due to an update. Changed it!